### PR TITLE
Allow reordering of turns in the Preset Editor

### LINKED ISSUE #24: Improve preset editor
Being able to add turns in between other turns would be nice, in case you forgot one and would have to delete everything after that, then re-add it to fix it.

### PR DESCRIPTION
**Pull request by lalitpatel**
### Summary

Adds ability to reorder turns in the preset editor.

In case the admin forgot to add a turn and wanted to add a turn in the middle, he would no longer have to delete everything after that, then re-add it to fix it. He can now just add a new row and drag it to its desired position.

Fixes #24

### Screenshots

demo

### CODE REVIEW

File: src/components/PresetEditor/PresetEditorTurn.tsx [hunk 1]
```diff
@@ -6,12 +6,15 @@ import Turn from "../../models/Turn";
...
     turn: Turn,
+    className: string,
     onValueChange: (turn: Turn | null, index: number) => void,
 }
 
-export const PresetEditorTurn = ({index, turn, onValueChange}: IProps) =>
-    <div className="columns is-mobile preset-editor-row" key={'turn-' + index}>
-        <div className="column is-1 has-text-vcentered is-size-5 has-text-grey">{index + 1}</div>
+export const PresetEditorTurn = ({index, turn, onValueChange, className, ...otherProps}: IProps) =>
+    <div className={className} {...otherProps}>
```
[the remark this answers is not in the document]
Comment (lalitpatel) [reply src/components/PresetEditor/PresetEditorTurn.tsx:14]: The library `react-sortablejs` adds a few props during runtime to the draggable element to make the drag drop work. This works fine if we have a native HTML DOM Element but not if we use a custom element unless we pass those props to the element below. 

An alternative to this is to not have the outermost `<div>` as a part of `<PresetEditorTurn>` component. But I figured that would mean that we didn't encapsulate all the presentation for that component within that component.

[the remark this answers is not in the document]
Comment (HSZemi) [reply src/components/PresetEditor/PresetEditorTurn.tsx:14]: Alrighty, thanks for the explanation :+1:

[the remark this answers is not in the document]
Comment (lalitpatel) [reply src/components/PresetEditor/PresetEditorTurn.tsx:14]: This is the alternative:

src/components/PresetEditor/PresetEditor.tsx
```diff
@@ -26,7 +26,7 @@
 import {ReactSortable} from "react-sortablejs";
 import {PresetEditorTurn} from "./PresetEditorTurn";
 
-interface Props extends WithTranslation{
+interface Props extends WithTranslation {
     preset: Preset | null,
     onSetEditorPreset: (preset: Preset) => ISetEditorPreset,
     onValueChange: (turn: Turn | null, index: number) => ISetEditorTurn,
@@ -49,10 +49,12 @@
         }
 
         const turns = this.props.preset.turns.map((turn: Turn, index: number) =>
-            <PresetEditorTurn index={index}
-                              turn={turn}
-                              className="columns is-mobile preset-editor-row"
-                              onValueChange={this.props.onValueChange} key={turn.id}/>);
+            <div className='columns is-mobile preset-editor-row'>
+                <PresetEditorTurn index={index}
+                                  turn={turn}
+                                  className=""
+                                  onValueChange={this.props.onValueChange} key={turn.id}/>
+            </div>);
 
         const presetCivilisations = this.props.preset.civilisations;
 
@@ -74,9 +76,12 @@
                     <div>
                         <div className="columns is-mobile has-text-weight-bold table-header">
                             <div className="column is-1 has-text-centered">#</div>
-                            <div className="column has-text-centered"><Trans i18nKey="presetEditor.host">Host</Trans></div>
-                            <div className="column has-text-centered"><Trans i18nKey="presetEditor.admin">Admin</Trans></div>
-                            <div className="column has-text-centered"><Trans i18nKey="presetEditor.guest">Guest</Trans></div>
+                            <div className="column has-text-centered"><Trans i18nKey="presetEditor.host">Host</Trans>
+                            </div>
+                            <div className="column has-text-centered"><Trans i18nKey="presetEditor.admin">Admin</Trans>
+                            </div>
+                            <div className="column has-text-centered"><Trans i18nKey="presetEditor.guest">Guest</Trans>
+                            </div>
                             <div className="column is-1"/>
                         </div>
```

src/components/PresetEditor/PresetEditorTurn.tsx
```diff
@@ -10,8 +10,8 @@
     onValueChange: (turn: Turn | null, index: number) => void,
 }
 
-export const PresetEditorTurn = ({index, turn, onValueChange, className, ...otherProps}: IProps) =>
-    <div className={className} {...otherProps}>
+export const PresetEditorTurn = ({index, turn, onValueChange}: IProps) =>
+    <>
         <div className="column is-1 has-text-vcentered is-size-5 has-text-grey has-text-left">
             <i className="material-icons has-text-grey has-cursor-grab is-drag-handle is-size-3">drag_indicator</i>
             {index + 1}</div>
@@ -29,4 +29,4 @@
                 onValueChange(null, index);
             }}/>
         </div>
-    </div>;
+    </>;
```

**Comment by HSZemi** [comment]
Thanks :yellow_heart:

**Pull request merged**